### DISCUSSION TITLE
"Calibrate" function ??

### DISCUSSION THREAD

**Discussion by softfoot** [OpenHantek/OpenHantek6022#230] [Q&A] [answered]
Hi, I love this software but there doesnt seem to be a "callibrate" function --- or am I missing something ??

I have a small DC offset on channel 2 and I was always able to fix it with the calibrate function in the original software.

Is there a way of doing this under OH ??

Best regards,
Dave

**Comment by Ho-Ro** [comment, marked as answer]
[RTFM](https://github.com/OpenHantek/OpenHantek6022#features) :)
The program uses offset/gain values from a config file - if this doesn't exist it takes the values from the EEPROM, either the factory offset calibration values or offset and gain values that you can create with an extra calibration program. The self-created values allow even sub-sample offset correction (due to oversampling).
More detailed info [can be found](https://github.com/Ho-Ro/Hantek6022API#create-calibration-values-for-openhantek) in the [firmware repo](https://github.com/Ho-Ro/Hantek6022API).

**Comment by softfoot** [comment]
Re RTFM - fair comment ;-)
    I looked at the instructions for installing the calibration
      program and I have python3 and fetched the libusb dll, but the
      instructions for "prepare Hantek6022API:" are only for linux
        and I don't have a linux box - so how do I do this last stage.
    Dave
    
      
    On 08/10/2021 18:45, Martin wrote:
    
    
      
      RTFM :)
        The program uses offset/gain values from a config file - if this
        doesn't exist it takes the values from the EEPROM, either the
        factory offset calibration values or offset and gain values that
        you can create with an extra calibration program. The
        self-created values allow even sub-sample offset correction (due
        to oversampling).
        More detailed info can be found in the firmware repo.
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view it on GitHub, or unsubscribe.
        Triage notifications on the go with GitHub Mobile for iOS or Android.
        
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/OpenHantek/OpenHantek6022/discussions/230#discussioncomment-1449746",
"url": "https://github.com/OpenHantek/OpenHantek6022/discussions/230#discussioncomment-1449746",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Reply by softfoot** [reply]
I gave up on trying to get "calibrate_6022.py" to run under Windows 7 - it was taking far too long, what with having to remember python on top of everything else.
In the end I used my programmable reference to supply the voltages and edited "modelDSO6022.conf" and restarted OpenHantek after each save.  This itterative aproach took about 30 minutes to get an acurate calibration setup - it's quite straightforward. The only drawback is that it isnt saved in the eeprom, but that just means you need to copy "modelDSO6022.conf" onto each machine before running "OpenHantek", hardly a chore.
Dave

**Reply by Ho-Ro** [reply]
Please check [this info about calibration with Windows](https://github.com/Ho-Ro/Hantek6022API/issues/8), it gives a step-by-step instruction - tested and confirmed. And as an extra benefit you get command line tools e. g. to record over long time.
Martin
